nbs/99_manuscript/coefs_comp/py/08_10-gtex_whole_blood-gene_pairs00.py

```diff
@@ -0,0 +1,459 @@
+    one files are found.
+
+    Args:
+        name: a string with the tissue name (or a part of it).
+
+    Returns:
+        A Path object pointing to the gene expression file for the
+        given tissue.
+    """
+    tissue_files = []
+    for f in TISSUE_DIR.glob("*.pkl"):
+        if name in f.name:
+            tissue_files.append(f)
+
+    assert len(tissue_files) == 1
+    return tissue_files[0]
+
+
+# %%
+# testing
+_tmp = get_tissue_file("whole_blood")
+assert _tmp.exists()
+
+
+# %%
+def plot_gene_pair(
+    tissue_name, gene0, gene1, hue=None, kind="hex", ylim=None, bins="log"
+):
+    """
+    It plots (joint plot) a gene pair from the given tissue. It saves the plot
+    for the manuscript.
+    """
+    # merge gene expression with metadata
+    tissue_file = get_tissue_file(tissue_name)
+    tissue_data = pd.read_pickle(tissue_file).T[[gene0, gene1]]
+    tissue_data = pd.merge(
+        tissue_data,
+        gtex_metadata,
+        how="inner",
+        left_index=True,
+        right_index=True,
+        validate="one_to_one",
+    )
+
+    # get gene symbols
+    gene0_symbol, gene1_symbol = gene_map[gene0], gene_map[gene1]
+    display((gene0_symbol, gene1_symbol))
+
+    # compute correlations for this gene pair
+    _clustermatch = cm(tissue_data[gene0], tissue_data[gene1])
+    _pearson = pearsonr(tissue_data[gene0], tissue_data[gene1])[0]
+    _spearman = spearmanr(tissue_data[gene0], tissue_data[gene1])[0]
+
+    _title = f"{tissue_name}\n$c={_clustermatch:.2f}$  $p={_pearson:.2f}$  $s={_spearman:.2f}$"
+
+    other_args = {
+        "kind": kind,  # if hue is None else "scatter",
+        "rasterized": True,
+    }
+    if hue is None:
+        other_args["bins"] = bins
+    else:
+        other_args["hue_order"] = ["Male", "Female"]
+
+    with sns.plotting_context("paper", font_scale=1.5):
+        p = sns.jointplot(
+            data=tissue_data,
+            x=gene0,
+            y=gene1,
+            hue=hue,
+            **other_args,
+            # ylim=(0, 500),
+        )
+
+        if ylim is not None:
+            p.ax_joint.set_ylim(ylim)
+
+        gene_x_id = p.ax_joint.get_xlabel()
+        gene_x_symbol = gene_map[gene_x_id]
+        p.ax_joint.set_xlabel(f"{gene_x_symbol}", fontstyle="italic")
+
+        gene_y_id = p.ax_joint.get_ylabel()
+        gene_y_symbol = gene_map[gene_y_id]
+        p.ax_joint.set_ylabel(f"{gene_y_symbol}", fontstyle="italic")
+
+        p.fig.suptitle(_title)
+
+        # save
+        output_file = (
+            OUTPUT_FIGURE_DIR
+            / f"gtex_{tissue_name}-{gene_x_symbol}_vs_{gene_y_symbol}.pdf"
+        )
+        display(output_file)
+
+        plt.savefig(
+            output_file,
+            bbox_inches="tight",
+            dpi=300,
+            facecolor="white",
+        )
+
+    return tissue_data
+
+
+# %% [markdown]
+# ## In whole blood (where this gene pair was found)
+
+# %%
+_tissue_data = plot_gene_pair(
```

Comment:
Exactly, this is one of the main findings: *UTY* (chr Y) and *KDM6A* (chr X). There is a linear relationship, but since female samples do not have *UTY*, the relationship is hidden when using Pearson or Spearman. It replicates in all tissues but female-specific ones (that's why we only see a single color, like in testis also).